src/main/java/net/snowflake/ingest/streaming/internal/BlobMetadata.java

```diff
@@ -49,16 +49,14 @@ List<ChunkMetadata> getChunks() {
     return this.chunks;
   }
 
-  /**
-   * Create {@link BlobMetadata} in case of {@link Constants.BdecVersion#ONE} and {@link
-   * BlobMetadataWithBdecVersion} otherwise to send BDEC version to server side.
-   */
+  @JsonProperty("bdec_version")
+  byte getVersionByte() {
+    return bdecVersion.toByte();
+  }
+
+  /** Create {@link BlobMetadata}. */
   static BlobMetadata createBlobMetadata(
       String path, String md5, Constants.BdecVersion bdecVersion, List<ChunkMetadata> chunks) {
-    // TODO SNOW-659721: Unify BlobMetadata with BlobMetadataWithBdecVersion once server side
```

Comment:
Looks like there're more places to be updated if you search for SNOW-659721?

Reply:
It looks that the JIRA was wrongly added in this context here.
SNOW-659721 is about parquet scanner on the server side,
the other comments, you can still find, are still relevant but not related to this PR.